bridge/eth/client.go

```diff
@@ -61,30 +61,24 @@ type Client struct {
 	gasHandler       bridge.GasHandler
 }
 
-func NewClient(
-	config bridge.EthereumConfig,
-	broadcaster bridge.Broadcaster,
-	mapper bridge.Mapper,
-	gasHandler bridge.GasHandler,
-) (*Client, error) {
+// ArgsClient is the DTO used in the client constructor
+type ArgsClient struct {
+	Config      bridge.EthereumConfig
+	Broadcaster bridge.Broadcaster
+	Mapper      bridge.Mapper
+	GasHandler  bridge.GasHandler
+	EthClient   *ethclient.Client
```

Comment:
You used pointers. There is no way we can ever use mocked implementations in integration tests.
Let's switch to:
```GO
EthClient     BlockchainClient
EthInstance BridgeContract
```

valid also for the ArgsRelayer